code/modules/robotics/bot/chefbot.dm

```diff
@@ -19,17 +17,21 @@
 	. = ..()
 	if (raging)
 		return
-	if(prob(60) && src.on == 1)
+	if(prob(src.emagged * 20))
+		drama()
+	if(!GET_COOLDOWN(src, "chefbot_yelling") && src.on)
+		SPAWN (0 SECOND)
+			if (src.emagged)
+				ON_COOLDOWN(src, "chefbot_yelling", pick(10 SECONDS, 20 SECONDS))
+			else
+				ON_COOLDOWN(src, "chefbot_yelling", pick(40 SECONDS, 60 SECONDS))
+			yell()
+	if(!GET_COOLDOWN(src, "chefbot_wander") && src.on)
 		src.navigate_to(get_step_rand(src))
-
-		SPAWN(0)
-			if(prob(src.emagged * 20))
-				drama()
-			if(prob(30 + src.emagged * 30))
-				yell()
+		ON_COOLDOWN(src, "chefbot_wander", pick(2 SECONDS, 5 SECONDS))
```

Comment:
Consider using `!ON_COOLDOWN` as the resolution of the statement removes the need for the subsequent use.  Just shift order of operations to check `src.on` first to short circuit (though preferably pull out `src.on` condition because it is applicable to each decision)

Reply:
Done, but only for the "chefbot_wander" cooldown. Since "chefbot_yelling"'s cooldown depends on a check, i left it as is.